components/eamxx/src/physics/mam/eamxx_mam_optics_process_interface.cpp

```diff
@@ -1,110 +1,531 @@
-  add_field<Computed>("aero_tau_sw", scalar3d_swband_layout, nondim, grid_name);
-  // longwave aerosol optical depth [-]
+  // 3D layout for shortwave aerosol fields: columns, number of shortwave, and nlev +1 s
+  // shortwave aerosol fields in mam has one extra level (nlev+1)
+  FieldLayout scalar3d_swband_layout_levp1{{COL, SWBND, ILEV},
+                                      {ncol_, nswbands_, nlev_ + 1}};
+
+  // 3D layout for shortwave aerosol fields: columns, number of shortwave, and nlev
+  FieldLayout scalar3d_swband_layout{{COL, SWBND, LEV},
+                                       {ncol_, nswbands_, nlev_}};
+
+  // 3D layout for longwave aerosol fields: columns, number of shortwave, and nlev +1 s
+  FieldLayout scalar3d_lwband_layout{{COL, LWBND, LEV},
+                                     {ncol_, nlwbands_, nlev_}};
+
+  FieldLayout scalar3d_layout_int{{COL, ILEV}, {ncol_, nlev_ + 1}};
+
+  // layout for 2D (1d horiz X 1d vertical) variable
+  FieldLayout scalar2d_layout_col{{COL}, {ncol_}};
+
+  // layout for 3D (2d horiz X 1d vertical) variables
+  FieldLayout scalar3d_layout_mid{{COL, LEV}, {ncol_, nlev_}};
+  add_field<Required>("omega", scalar3d_layout_mid, Pa / s,
+                      grid_name);  // vertical pressure velocity
+  add_field<Required>("T_mid", scalar3d_layout_mid, K,
+                      grid_name);  // Temperature
+  add_field<Required>("p_mid", scalar3d_layout_mid, Pa,
+                      grid_name);  // total pressure
+
+  add_field<Required>("p_int", scalar3d_layout_int, Pa,
+                      grid_name);  // total pressure
+  add_field<Required>("pseudo_density", scalar3d_layout_mid, Pa, grid_name);
+  add_field<Required>("pseudo_density_dry", scalar3d_layout_mid, Pa, grid_name);
+
+  add_field<Required>("qv", scalar3d_layout_mid, q_unit, grid_name,
+                      "tracers");  // specific humidity
+  add_field<Required>("qi", scalar3d_layout_mid, q_unit, grid_name,
+                      "tracers");  // ice wet mixing ratio
+  add_field<Required>("ni", scalar3d_layout_mid, n_unit, grid_name,
+                      "tracers");  // ice number mixing ratio
+
+  // droplet activation can alter cloud liquid and number mixing ratios
+  add_field<Required>("qc", scalar3d_layout_mid, q_unit, grid_name,
+                     "tracers");  // cloud liquid wet mixing ratio
+  add_field<Required>("nc", scalar3d_layout_mid, n_unit, grid_name,
+                     "tracers");  // cloud liquid wet number mixing ratio
+
+  add_field<Required>("phis", scalar2d_layout_col, m2 / s2, grid_name);
+  add_field<Required>("cldfrac_tot", scalar3d_layout_mid, nondim,
+                      grid_name);  // cloud fraction
+  add_field<Required>("pbl_height", scalar2d_layout_col, m,
+                      grid_name);  // planetary boundary layer height
+
+  // shortwave aerosol scattering asymmetry parameter [unitless]
+  add_field<Computed>("aero_tau_g_sw_mam4", scalar3d_swband_layout_levp1, nondim,
```

Comment:
I think these _mam4 fields can be removed from computed.

Reply:
I removed the _mam4 filed from computed; they are temporal view now. Should we also remove `aero_tau_forward` (aerosol forward scattered fraction * tau * w)? rrtmgp does not required this filed, but I do not know if this variables is needed for diagnostics. @mahf708 @brhillman

Reply:
Okay, I will remove `aero_tau_forward` from the FM.